### PR TITLE
DDF-4105 Prevent parsing error on empty date strings

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
Prevent parsing error on empty date strings
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@rzwiefel
#### How should this be tested?
Full Build / Edge case covered in unit tests

#### What are the relevant tickets?
[](https://codice.atlassian.net/browse/)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java [hunk 1]
```diff
@@ -584,6 +585,10 @@ public Serializable parseDate(Serializable value) {
       return null;
     }
 
+    if (StringUtils.isEmpty(value)) {
```
Comment (GabrielFabian) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: ✏️ nit: could combine this check with the check on 584. It's just style though, some prefer to be more explicit. Approved either way.

```
if (!(value instanceof String) || StringUtils.isEmpty(value)) {
      return null;
    }
```

Comment (rzwiefel) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: I'm biased for this code 😂 but I do for this particular case like the explicit separate statements. (primarily because of the inverted first conditional - with that plus adding in an OR i think it adds a nontrivial amount of processing needed to read/process this code)

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java [hunk 2]
```diff
@@ -93,6 +93,7 @@
 import org.slf4j.LoggerFactory;
 import spark.Request;
 import spark.Response;
+import spark.utils.StringUtils;
```
Comment (coyotesqrl) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: ❓ Curious why we're using these `StringUtils` and not Apache commons?

Comment (bdeining) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/util/EndpointUtil.java]: that was an accident; fixed

**Comment by bdeining** [comment]
build now
